mantle/kola/harness.go

```diff
@@ -924,10 +974,103 @@ func RegisterExternalTests(dir string) error {
 	return RegisterExternalTestsWithPrefix(dir, basename)
 }
 
+// Create a parent test that runs non-exclusive tests as subtests
+func makeNonExclusiveTest(tests []*register.Test, config *conf.UserData, flight platform.Flight) (register.Test, func(h *harness.H)) {
+	// Parse test flags
+	var (
+		internetAccess     bool
+		noSSHKeyInMetadata bool
+		noSSHKeyInUserdata bool
+	)
+	var flags []register.Flag
+	for _, test := range tests {
+		if !noSSHKeyInMetadata && test.HasFlag(register.NoSSHKeyInMetadata) {
+			flags = append(flags, register.NoSSHKeyInMetadata)
+			noSSHKeyInMetadata = true
+		}
+		if !noSSHKeyInUserdata && test.HasFlag(register.NoSSHKeyInUserData) {
+			flags = append(flags, register.NoSSHKeyInUserData)
+			noSSHKeyInUserdata = true
+		}
+		if !internetAccess && testRequiresInternet(test) {
+			flags = append(flags, register.RequiresInternetAccess)
+			internetAccess = true
+		}
+	}
+
+	nonExclusiveWrapper := register.Test{
+		Name: "non-exclusive",
+		Run: func(c cluster.TestCluster) {
+			// noop
+			_ = c
+		},
+		UserData: config,
+		// This will allow runTest to copy kolet to machine
+		NativeFuncs: make(map[string]register.NativeFuncWrap),
+		ClusterSize: 1,
+		Flags:       flags,
+	}
+
+	runNonExclusive := func(parenth *harness.H) {
```

Comment:
`parenth` I assume means parent harness ? Maybe `parentH` would be more obvious?